src/state/opamUpdate.ml

```diff
@@ -75,7 +74,8 @@ let repository rt repo =
     else match eval_redirect gt r repo_root with
       | None -> Done (r, has_changes)
       | Some (new_url, f) ->
-        OpamFilename.cleandir repo_root;
+        OpamRepositoryRoot.remove repo_root;
+        OpamRepositoryRoot.make_empty repo_root;
```

Comment:
Do we need to create an empty directory here ? `OpamHTTP` and `OpamVCS` `fetch_repo_update` does not need the directory to exist, but `OpamLocal` requires it. I think it should be in there that the change should be applied. Plus, it'll make the functions more homogeneous.